internal/stats/aggregate.go

```diff
@@ -0,0 +1,279 @@
+		groupedMetricBuckets[bucketKey] = append(groupedMetricBuckets[bucketKey], stringMetric)
+	}
+
+	return groupedMetricBuckets, nil
+}
+
+func hasAllNeededFields(metric map[string]any, query AggregationQuery) bool {
+	for _, field := range query.GroupByFields {
+		_, exists := metric[field]
+		if !exists {
+			return false
+		}
+	}
+
+	_, exists := metric[query.OverviewField]
+	if !exists {
+		return false
+	}
+
+	return true
+}
+
+// Calculate aggregations for a group of metrics.
+func computeGroupAggregation(groupedMetrics []map[string]string, query MetricQuery) (map[string]any, error) {
+	if len(groupedMetrics) == 0 {
+		return nil, nil
+	}
+
+	aggregatedResults, err := extractFields(groupedMetrics[0], query)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to extract group by fields from metric '%v': '%v'.", groupedMetrics[0], err)
+	}
+
+	// Determine if aggregation should be numeric or non-numeric.
+	isNumeric := canAggregateAsNumeric(groupedMetrics[0], query.OverviewField)
+
+	if isNumeric {
+		return aggregateNumeric(groupedMetrics, aggregatedResults, query.OverviewField)
+	} else {
+		return aggregateNonNumeric(groupedMetrics, aggregatedResults)
+	}
+}
+
+func extractFields(metric map[string]string, query MetricQuery) (map[string]any, error) {
+	aggregatedResults := make(map[string]any)
+	groupByValues := make(map[string]string)
+
+	for _, field := range query.GroupByFields {
+		value := metric[field]
+		groupByValues[field] = value
+	}
+
+	aggregatedResults[GROUP_BY] = groupByValues
+	aggregatedResults[OVERVIEW] = query.OverviewField
+
+	return aggregatedResults, nil
+}
+
+func canAggregateAsNumeric(metric map[string]string, overviewField string) bool {
+	aggregateValue := metric[overviewField]
+	_, err := util.StrToFloat(aggregateValue)
+	if err == nil {
+		return true
+	}
+
+	return false
+}
+
+func aggregateNumeric(groupedMetrics []map[string]string, aggregateMap map[string]any, overviewField string) (map[string]any, error) {
+	var values []float64
+	statsMap := make(map[string]any)
+
+	for _, metric := range groupedMetrics {
+		aggregateValue := metric[overviewField]
+
+		numericValue, err := util.StrToFloat(aggregateValue)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to convert string to float: '%v'.", err)
+		}
+
+		values = append(values, numericValue)
+	}
+
+	stats := util.ComputeAggregates(values)
+	statsJSON, err := util.ToJSON(stats)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to convert stats to JSON: '%v'.", err)
+	}
+
+	statsJSONMap, err := util.JSONMapFromString(statsJSON)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to convert JSON string to JSON map: '%v'.", err)
+	}
+
+	for aggregateKey, aggregateValue := range statsJSONMap {
+		statsMap[aggregateKey] = aggregateValue
+	}
+
+	aggregateMap[STATS] = statsMap
+
+	return aggregateMap, nil
+}
+
+func aggregateNonNumeric(groupedMetrics []map[string]string, aggregateMap map[string]any) (map[string]any, error) {
```

Comment:
This should call into util, so we can expand it in the future (e.g., most common string, length stats, etc).